### PR TITLE
ivy: add sys 'read' 'file' for reading files

### PR DESCRIPTION
**Pull request by rsc**
sys 'read' 'file' returns a vector of file lines.
If file contains a matrix, then ivy@ sys 'read' 'file' converts the lines to a matrix.

### CODE REVIEW

**Comment by rsc** [comment]
The test depends on the @ change, so that should be submitted first (or the test should be removed).

File: value/sys.go [hunk 1]
```diff
@@ -39,53 +43,134 @@ To convert a time vector to a seconds value:
 To print seconds in Unix date format:
```
Comment (robpike) [inline value/sys.go:43]: Need to add a one-line description of the new feature in sysHelp

File: value/sys.go [hunk 2]
```diff
@@ -39,53 +43,134 @@ To convert a time vector to a seconds value:
...
+	v, ok := args[0].(Vector)
+	if !ok || !allChars(v) {
+		usage()
+	}
+	file := vecText(v)
+
+	f, err := os.Open(file)
+	if err != nil {
+		Errorf("%v", err)
+		return empty
```
Comment (robpike) [inline value/sys.go:156]: you don't need this return,

File: value/sys.go [hunk 3]
```diff
@@ -39,53 +43,134 @@ To convert a time vector to a seconds value:
...
+	}
+	defer f.Close()
+
+	var out Vector
+	r := bufio.NewReader(f)
+	for {
```
Comment (robpike) [inline value/sys.go:162]: for minor consistency, i'd prefer you to use Scanner over Reader.

File: value/sys.go (hunk 2 above)
Comment (rsc) [reply value/sys.go:156]: Done.

File: value/sys.go (hunk 3 above)
Comment (rsc) [reply value/sys.go:162]: Done.

File: value/sys.go (hunk 1 above)
Comment (rsc) [reply value/sys.go:43]: Done.

**Comment by rsc** [comment]
Pushed an update. Still depends on landing the @ change first.

File: value/sys.go [hunk 4]
```diff
@@ -39,53 +43,129 @@ To convert a time vector to a seconds value:
...
+		}
+		if fn, ok := sys2[verb]; ok {
+			return fn(c, []Value{})
+		}
+		Errorf("sys %q not defined", verb)
+	}
+
+	if v1, ok := vv[0].(Vector); ok && allChars(v1) { // multiple arguments, verb first
+		verb := vecText(v1)
+		if fn, ok := sys2[verb]; ok {
```
Comment (robpike) [inline value/sys.go:68]: maybe sysN or sysV for future-proofing?

File: value/sys.go [hunk 5]
```diff
@@ -39,53 +43,129 @@ To convert a time vector to a seconds value:
...
-	switch arg[1 : len(arg)-1] {
-	case "help":
-		fmt.Fprint(conf.Output(), sysHelp)
+
+	Errorf("sys requires string argument")
+	panic("unreachable")
+}
+
+var sys1 = map[string]func(c Context) Value{
+	"help": func(c Context) Value {
```
Comment (robpike) [inline value/sys.go:82]: conf := c.Config() will make the code (and the changes) notably shorter.

**Review by robpike — CHANGES_REQUESTED** [review]

File: value/sys.go (hunk 4 above)
Comment (rsc) [reply value/sys.go:68]: Done.

File: value/sys.go (hunk 5 above)
Comment (rsc) [reply value/sys.go:82]: Done, although I think it ends up longer, because `conf *config.Config; conf` is longer than `c Context; c.Config()`.

**Pull request merged**